server/tests/test_background_tasks.py

```diff
@@ -454,7 +474,8 @@ def num_completed_tasks():
         worker.terminate()
 
     expected_sorted_results = list(range(num_tasks))
+    results = [
+        num for num, in db_session.execute("SELECT num FROM count_results").fetchall()
+    ]
     # Each task should have run exactly once
     assert sorted(results) == expected_sorted_results
-    # Tasks should not have run in order, since some got interrupted and reset
-    assert results != expected_sorted_results
```

Comment:
While I like what this assertion is trying to test, it relies on non-deterministic timing for the tasks to get executed out of order. I couldn't think of an easy way to make this work deterministically, so to prevent flaky tests I'm removing it.